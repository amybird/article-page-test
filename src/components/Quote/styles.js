import styled from 'styled-components';
import mq from '../../styles/mediaQueries';

export const QuoteText = styled.p`
    font-size:30px;
    line-height: 40px;
    width:100%;
    padding: 10px;
    text-align: left;
    text-decoration:underline;

    ${mq.medium`
        font-size:40px;
        line-height: 50px;
    `}
`;
