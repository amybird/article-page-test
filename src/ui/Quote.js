import React from 'react';
import styled, {css} from 'styled-components/macro';
import mq from '../styles/mediaQueries';

const Quote = ({quote, ...props}) => (
    <QuoteText>{props[0] ? props[0].quote : quote}</QuoteText>
)

const QuoteText = styled.p`
    font-size:30px;
    line-height: 40px;
    width:100%;
    padding: 10px;
    text-align: left;
    text-decoration:underline;
    ${mq.medium(css`
        font-size:40px;
        line-height: 50px;
    `)}
`;

export default Quote;