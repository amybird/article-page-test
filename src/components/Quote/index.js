import React from 'react';
import PropTypes from 'prop-types';
import { QuoteText } from './styles';

const Quote = ({ text }) => (
  <QuoteText>{text}</QuoteText>
);

Quote.propTypes = {
  text: PropTypes.string.isRequired
};

export default Quote;
