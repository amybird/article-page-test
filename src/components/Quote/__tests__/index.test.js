import React from 'react';
import renderer from 'react-test-renderer';
import Quote from '../index';

describe('Quote', () => {
  it('should render a quote', () => {
    const tree = renderer
      .create(<Quote text="Testing a quote" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
