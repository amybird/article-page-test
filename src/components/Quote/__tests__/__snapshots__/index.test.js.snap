// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Quote should render a quote 1`] = `
.c0 {
  font-size: 30px;
  line-height: 40px;
  width: 100%;
  padding: 10px;
  text-align: left;
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

@media (min-width:768px) {
  .c0 {
    font-size: 40px;
    line-height: 50px;
  }
}

<p
  className="c0"
>
  Testing a quote
</p>
`;
